### PR TITLE
group_split(keep=) -> group_split(.keep=)

### PR DESCRIPTION
**Pull request by romainfrancois**
probably also needs `group_map()` and `group_modify()`

### CODE REVIEW

**Comment by romainfrancois** [comment]
Should we still have a `keep = `, perhaps warning if used ? 

This family of functions is experimental, but this might anyway break some packages.

**Comment by romainfrancois** [comment]
fixes #5174

File: R/group_map.R [hunk 1]
```diff
@@ -115,17 +115,17 @@ as_group_map_function <- function(.f) {
 #'   group_modify(~ head(.x, 2L))
 #'
 #' @export
-group_map <- function(.data, .f, ..., keep = FALSE) {
+group_map <- function(.data, .f, ..., .keep = FALSE) {
```
Comment (DavisVaughan) [inline R/group_map.R:118]: Update roxygen2 docs to `.keep`?

File: tests/testthat/test-group_split.R [hunk 2]
```diff
@@ -12,7 +12,7 @@ test_that("group_split() keeps the grouping variables by default", {
 
 test_that("group_split() can discard the grouping variables with keep = FALSE", {
```
Comment (DavisVaughan) [inline tests/testthat/test-group_split.R:13]:
```suggestion
test_that("group_split() can discard the grouping variables with .keep = FALSE", {
```

File: R/group_map.R (hunk 1 above)
Comment (romainfrancois) [reply R/group_map.R:118]: Thanks.

File: R/group_split.R [hunk 3]
```diff
@@ -66,46 +66,58 @@
...
-group_split <- function(.tbl, ..., keep = TRUE) {
+group_split <- function(.tbl, ..., .keep = TRUE) {
   UseMethod("group_split")
 }
 
 #' @export
-group_split.data.frame <- function(.tbl, ..., keep = TRUE) {
+group_split.data.frame <- function(.tbl, ..., .keep = TRUE, keep = deprecated()) {
+  if (!missing(keep)) {
+    lifecycle::deprecate_warn("1.0.0", "group_split(keep = )", "group_split(.keep = )")
```
Comment (romainfrancois) [inline R/group_split.R:76]: Should this rather be in the generic ?

**Pull request merged**